app/models/facilities_query.rb

```diff
@@ -0,0 +1,178 @@
+# frozen_string_literal: true
+
+class FacilitiesQuery
+  attr_reader :params
+
+  def initialize(params)
+    @params = params
+  end
+
+  def query
+    if valid_location_query?
+      location_query_klass.new(params).run
+    elsif @params[:ids]
+      build_result_set_from_ids(params[:ids]).flatten
+    else
+      BaseFacility.none
+    end
+  end
+
+  # Massages the location_query_klass return value to be boolean
+  def valid_location_query?
+    !location_query_klass.nil?
+  end
+
+  # When given more than one type of distance query param,
+  # return nil because the app will not choose preference for the user.
+  # In the controller, this logic is somewhat duplicated
+  # and will render an error when given multiple params.
+  # In the case that only one of these types is given,
+  # return the class used to make that type of query.
+  def location_query_klass
+    @location_query_klass ||= case location_keys
+                              when [] then nil
+                              when %i[lat long] then RadialQuery
+                              when [:state]     then StateQuery
+                              when [:zip]       then ZipQuery
+                              when [:bbox]      then BoundingBoxQuery
+                              end
+  end
+
+  def location_keys
+    (%i[lat long state zip bbox] & params.keys.map(&:to_sym)).sort
+  end
+
+  def build_result_set_from_ids(ids)
+    ids_for_types(ids).map do |facility_type, unique_ids|
+      klass = "Facilities::#{facility_type.upcase}Facility".constantize
+      klass.where(unique_id: unique_ids)
+    end
+  end
+
+  def ids_for_types(ids)
+    ids.split(',').each_with_object({}) do |type_id, obj|
+      facility_type, unique_id = type_id.split('_')
+      if facility_type && unique_id
+        obj[facility_type] ||= []
```

Comment:
above you could initialize the memo object as a hash with default of an array so this line can be eliminated
```
ids.split(',').each_with_object(Hash.new{|hash,key| hash[key] = []}) do |type_id, obj|
```

i've never seen `each_with_object` before!  that's awesome!

Reply:
@erluti rubocop will actually change inject to each_with_object -- it was new to me when I came on to the project as well!

Reply:
Yeah I ran a `make ci` locally and all kinds of weird things happened 😭

Reply:
I run `guard` as I develop and it's less shocking than having it build up over time :)